source/deployment/cluster.rst

```diff
@@ -1,116 +1,116 @@
 
-- `Deployment Guide <https://docs.mattermost.com/deployment/cluster.html#id1>`_ - **Instructions to set up and maintain high availability on your Mattermost servers.**
- - `Initial Setup Guide for High Availability <https://docs.mattermost.com/deployment/cluster.html#id2>`_
- - `Adding a Server to the Cluster <https://docs.mattermost.com/deployment/cluster.html#id4>`_
- - `Removing a Server from the Cluster <https://docs.mattermost.com/deployment/cluster.html#id9>`_
+You can apply most configuration changes and dot release security updates without interrupting service, provided that you update the system components in the correct sequence. Exceptions to this rule are changes to configuration settings that require a server restart, and updates that involve a change to the database schema. Downtime for a server restart is around 5 seconds, and for a database schema update, downtime is around 30 seconds.
 
-- `Configuration and Compatibility <https://docs.mattermost.com/deployment/cluster.html#id13>`_ - **Details on configuring your system for high availability.**
- - `Mattermost Server <https://docs.mattermost.com/deployment/cluster.html#mattermost-server-configuration>`_ - Configure the cluster settings for the Mattermost application server. 
- - `Proxy Server <https://docs.mattermost.com/deployment/cluster.html#proxy-server-configuration>`_ - Configure NGINX or another load balancer to proxy servers in the cluster.
- - `File Storage Location <https://docs.mattermost.com/deployment/cluster.html#file-storage-configuration>`_ - Configure the file storage location to be compatible with a high availability setup.
- - `Database <https://docs.mattermost.com/deployment/cluster.html#database-configuration>`_ - Size and deploy a multi-database configuration for high availability and scaling.
+The following outline describes the procedure for updating a high availability system without a service interruption.
 
-- `Troubleshooting <https://docs.mattermost.com/deployment/cluster.html#id14>`_ - **Advice on troubleshooting your high availability setup.**
-
------
+1. **Disconnect redundant servers** - Configure load balancers to consolidate requests to a single application server capable of supporting anticipated traffic during the update process. Depending on the resources available, you might choose to run updates during low traffic periods.
+2. **Apply single server update** - Apply the update to a single server by updating ``config.json`` and running a command to load the new configuration into the application server, which continues to operate without down time.
+3. **Apply update to redundant servers** - Apply the same updates to the redundant servers.
+4. **Reconnect redundant servers** - Configure load balancers to once again balance traffic across all application servers.
```

Comment:
@it33 Should we document only the rolling config update? It seems faster, requires less intervention (ie, no consolidation to a single server), and (it seems to me at least) makes sizing less critical because you're not taking down all-but-one server.

@coreyhulen Are there configuration updates when a rolling update is not feasible?